### ISSUE TITLE
Compiler panic in llvm_backend_type.cpp

### ISSUE THREAD

**Issue by jasonKercher** [odin-lang/Odin#1442]
## Context

Odin: dev-2022-01:cfbc1a44
OS:   Arch Linux, Linux 5.15.13-arch1-1
CPU:  Intel(R) Core(TM) i5-2520M CPU @ 2.50GHz
RAM:  7833 MiB

Not sure how relevant, but compiled against llvm 13

## Expected Behavior

Program should compile

## Current Behavior

```sh
$> odin build .
src/llvm_backend_type.cpp(12): Panic: NOT FOUND lb_type_info_index Plan @ index 229
Illegal instruction (core dumped)
```

NOTE: There is a struct named `Plan` in the code.

## Failure Information (for bugs)

Based on the error, and the code, it looks like the `Plan` type does not exist in `minimum_dependency_type_info_set`.  Types are added via the function `add_min_dep_type_info` (checker.cpp:1822):

```cpp
void add_min_dep_type_info(Checker *c, Type *t)
```

Using gdb, I was able to determine that the Plan `type` never makes it into add_min_dep_type_info.  I'm not familiar enough with the compiler internals to understand *why* that happens though.

### Steps to Reproduce

Unfortunately, min rep is a bit long (~80 lines plus small package), so I made a repo for it: 
https://github.com/jasonKercher/compilerbug

Then run from within the repo:
```sh
odin build main.odin
```

**Comment by awwdev** [comment]
possibly related to https://github.com/odin-lang/Odin/issues/1267

**Comment by gingerBill** [comment]
Appears to be fixed when #1267 was fixed.
